llvm/lib/Target/SystemZ/SystemZTargetTransformInfo.cpp

```diff
@@ -1322,26 +1323,30 @@ SystemZTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
   return BaseT::getIntrinsicInstrCost(ICA, CostKind);
 }
 
+// Find the type of the vector operand indicated by index.
+// Asserts that the operand indicated is actually a vector.
+FixedVectorType *getOperandVectorType(const IntrinsicInst *II, unsigned Index) {
+  auto *T = II->getOperand(Index)->getType();
+  assert(T->isVectorTy());
+  return cast<FixedVectorType>(T);
+}
+
```

Comment:
Since the intrinsic is defined so that it must have a vector of integer values, we do not need to verify this. I think this should be the same for other intrinsics as well. Did you try to use some other kind of type and compile it with 'opt'? I think it will verify the module and report broken IR.

Given that we don't need to have an extra redundant check here, I think this method (at least for now) is a bit superfluous - probably easier to just get the type directly in one line below.

Reply:
The vectors can also have float values, but that would not preclude the `FixedVectorType`. The idea behind the assert was that when you use this function to get the operand type for a given intrinsics vector operand, to check that you supplied the correct operand index, so we don't run into the same error that motivated this PR in the first place.

Reply:
I think that is over-asserting as the operand index / type never changes and is trivially entered just once and has to be correct per the language spec.

The problem was that we handled some other intrinsic by accident, not that we messed up the operand index of the one we are handling, right?

Do you agree, or you think I am missing something here?

Reply:
I think handling other intrinsics was intended (but primarily for future use), and the problem arose because some of them had the vector operand in a different place.

But this is not a hill i'm willing to die on. If you prefer to not have the assert and then move the cast out of the function and back into the cast, i'll do that.

Reply:
It's kind of interesting if it is needed or not, and maybe also a matter of style, perhaps. @uweigand maybe has an opinion as well?

Reply:
I do believe this assert is redundant - it is fails, the immediately following cast would also fail, which would give us the same information.   And without the assert, the function becomes so trivial that it might as well be inlined back in place?

Reply:
correct, without the assert, the function can go.